src/lib/utils.ts

```diff
@@ -160,6 +167,57 @@ export function guessKeyTypeFromKeyName(
   return 'Singleton';
 }
 
+export const encodeTupleKeyValue = (
+  valueContent: string, // i.e. (bytes4,Number,bytes16)
+  valueType: string, // i.e. (bytes4,bytes8,bytes16)
+  decodedValues: Array<string | number | JSONURLDataToEncode>,
+) => {
+  // We assume data has already been validated at this stage
+
+  const valueTypeParts = valueType
+    .substring(1, valueType.length - 1)
+    .split(',');
+  const valueContentParts = valueContent
+    .substring(1, valueContent.length - 1)
+    .split(',');
+
+  if (valueTypeParts.length !== decodedValues.length) {
+    throw new Error(
+      `Can not encode tuple key value: ${decodedValues}. Expecte array of length: ${valueTypeParts.length}`,
+    );
+  }
+
+  return (
```

Comment:
Id suggest putting this inside a variable and returning that so its clear whats being returned. e.g. encodedTupleString